### PR TITLE
novelrt: Add recipe

### PR DESCRIPTION
**Pull request by capnkenny**
Specify library name and version: **novelrt/0.0.1**

This introduces the NovelRT library (https://github.com/novelrt/NovelRT) as a recipe allowing developers to create 2D games and Visual Novels (of which I am one of the maintainers).
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
see comment and below: https://github.com/conan-io/conan-center-index/pull/9063#issuecomment-1021175018

**Comment by SSE4** [comment]
@capnkenny I think it affects only `clang11 + libstdc++`, `clang11 + libc++` should be fine.
could you double-check the right standard library is used (in other words, `-stdlib=libc++` is applied)?
it seems like you're not using CMake wrapper, it might be the reason (it is currently needed, so conan can inject proper flags).

File: recipes/novelrt/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    def package(self):
+        self.copy("*LICENCE.md", dst="licenses", keep_path=False)
+        self.copy("include/NovelRT/*.h", dst="", keep_path=True)
+        self.copy("*include/NovelRT.h", dst="include", keep_path=False)
+        self.copy("*.spv", dst="bin/Resources/Shaders", keep_path=False)
+        if self.settings.os == "Windows":
+            self.copy("*NovelRT.lib", dst="lib", keep_path=False)
+            self.copy("*NovelRT.dll", dst="bin", keep_path=False)
+            self.copy("*bz2.dll", dst="bin", keep_path=False)
+            self.copy("*FLAC.dll", dst="bin", keep_path=False)
```
Comment (SSE4) [inline recipes/novelrt/all/conanfile.py:112]: it sounds weird... it shouldn't install DLLs of dependencies, like flac, ogg, vorbis, etc.

File: recipes/novelrt/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        return str(self.settings.compiler.runtime) in ["MDd", "MTd"]
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
+
+    def configure_cmake(self):
+        cmakePath = self.deps_env_info["cmake"].CMAKE_ROOT
+        if cmakePath:
+            if self.settings.os == "Windows":
+                cmakePath += "\\bin\\cmake.exe"
```
Comment (SSE4) [inline recipes/novelrt/all/conanfile.py:82]: shouldn't be necessary to manipulate with `cmake` paths here. conan should be able to automatically apply the right executable from build requirements.

File: recipes/novelrt/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            else:
+                cmakePath += "/bin/cmake"
+            self.output.info(f"Overriding CMake to use version from Conan at: {cmakePath}")
+        cmake = CMake(self, cmake_program=cmakePath, parallel=True)
+        cmake.definitions["NOVELRT_BUILD_DOCUMENTATION"] = "Off"
```
Comment (SSE4) [inline recipes/novelrt/all/conanfile.py:87]: just in case, you can use python `False` here

**Comment by capnkenny** [comment]
After talking with my team, it appears that both our side of things and what we'd like from CCI aren't _fully_ there yet, so we're settling on a different distribution method for now. Thank you for the solid advice though! We're taking the recommendations here and applying it to our Conanfile usage in hopes that we can revisit this in the future.

**Pull request closed without merging**